code/modules/mob/living/simple_animal/friendly/arcane_golem.dm

```diff
@@ -0,0 +1,82 @@
+/mob/living/simple_animal/arcane_golem
+	name = "arcane golem"
+	desc = "A construct created by wizards who wish to avoid human contact, but still want a companion. It's inexpensive, doesn't annoy its master and never refuses to learn a new spell."
+
+	icon = 'icons/mob/critter.dmi'
+	icon_state = "arcane_golem"
+	icon_living = "arcane_golem"
+	treadmill_speed = 0
+
+
+	health = 50
+	maxHealth = 50
+
+	speak = list("Servio.", "Magister?", "Ego enim arcana.")
+	speak_emote = list("vocalizes")
+
+	speak_chance = 0.25
+
+	var/spell/aoe_turf/conjure/arcane_golem/master_spell
+
+	min_oxy = 0
+	max_oxy = 0
+	min_tox = 0
+	max_tox = 0
+	min_co2 = 0
+	max_co2 = 0
+	min_n2 = 0
+	max_n2 = 0
+	heat_damage_per_tick = 0
+	cold_damage_per_tick = 0
+
+
+/mob/living/simple_animal/arcane_golem/Destroy()
+	..()
+
+	if(master_spell)
+		master_spell.golems.Remove(src)
+		master_spell = null
+
+/mob/living/simple_animal/arcane_golem/Life()
+	..()
+
+	if(!master_spell)
+		return
+	if(!master_spell.holder)
+		master_spell = null
+		return
+	if(ckey)
+		return
+
+	var/mob/master = master_spell.holder
+	if(istype(master))
+		if(master.isDead() || isnull(master.loc))
+			Die()
+			return
+
+		//Alive - follow the master
+		if(!isDead())
+			//Don't wander around if wizard is nearby - follow him instead
+			if(step_to(src, master, 2))
```

Comment:
step_to returns 0 if the movement failed (can't find a path, or the path's length is smaller than, in this case, 2). I didn't have any issues with this

also butt screw github for removing ability to make line comments on the phone